src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts

```diff
@@ -384,10 +374,10 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 					rejectButtonLabel: 'Abbrechen',
 				})
 			) {
-				this.badgeClassForm.controls.competencies.reset();
-				const controls = this.badgeClassForm.controls.competencies.controls;
-				for (let i = controls.length - 1; i >= 0; i--) {
-					this.badgeClassForm.controls.competencies.removeAt(controls.indexOf(controls[i]));
+				const competencies = this.badgeClassForm.controls.competencies;
+				competencies.reset();
+				for (let i = competencies.length - 1; i >= 0; i--) {
+					this.badgeClassForm.controls.competencies.removeAt(competencies[i]);
```

Comment:
I still don't understand, why you have to remove the entries one by one. If this is in fact necessary, maybe a comment explaining why would be nice. If not, surely there exists a method that removes them all (like `clear`)?
Even if not, I don't think this would work that way, because now you're calling `removeAt` with the competency, instead of the index. I think what should work would be:
```ts
for (let i = competencies.length - 1; i >= 0; i--) {
    competencies.removeAt(i);
}
```

Reply:
Did not find a method that would remove all at once, i guess the closes thing would be to call ```competencies.setValue([])```. Not sure if thats better than removing the competencies, for now i pushed the fix you suggested